redwood/api/src/functions/getVerifiedExternalAddresses/getVerifiedExternalAddresses.ts

```diff
@@ -0,0 +1,29 @@
+import {logger} from 'src/lib/logger'
+import {db} from 'src/lib/db'
+import {CachedProfile} from 'src/services/cachedProfiles/cachedProfiles'
+
+// This endpoint is used by the Snapshot strategy named `zorro`
+export const handler = async (event, context) => {
+  const params = JSON.parse(event.body)
+  const {purposeIdentifier, externalAddresses, snapshot} = params
+
+  const connections = await db.connection.findMany({
+    where: {
+      purposeIdentifier,
+      externalAddress: {in: externalAddresses.map((str) => str.toLowerCase())},
+    },
+    include: {
+      cachedProfile: true,
+    },
+  })
+
+  const verifiedExternalAddresses = connections
+    .filter((connection) => CachedProfile.isVerified(connection.cachedProfile))
```

Comment:
Ah, this is *slightly* scary to me. The `isVerified` and `currentStatus` logic is untested. I'd assumed that a bug there was nbd, since getting it wrong would just show the UI in an incorrect state/be somewhat confusing. If we're actually depending on it to give the right results for eg. a Snapshot vote (which totally makes sense, I just didn't make the mental connection) I should definitely add some tests.

Reply:
Oh also, you should just import the top-level `isVerified` function from `services/cachedProfiles.ts`, not the `CachedProfile` object. The `CachedProfile` object expects the arguments to come in a format compatible with the GraphQL resolver spec, so it doesn't take an unwrapped profile directly.

Reply:
on the latter point: yup noticed that the exports from `services/cachedProfile.ts` had changed after I tried testing this in prod (I hadn't tested the version merged into main locally), and fixed right after